From 2d3b42ca895e411fc57fce07e9dcf8a1cbbe276f Mon Sep 17 00:00:00 2001
From: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
Date: Sun, 18 Jan 2015 11:18:13 -0800
Subject: [PATCH 02/11] Define EXC_MASK_CRASH and MACH_EXCEPTION_CODES if
 they're not defined in the SDK

The 10.4 SDK does not define these macros

Signed-off-by: Jeremy Huddleston Sequoia <jeremyhu@apple.com>
---
 lib/Support/Unix/Signals.inc | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git llvm_release_38/lib/Support/Unix/Signals.inc macports_release_38/lib/Support/Unix/Signals.inc
index 061cdb3da21..4d687b304cc 100644
--- llvm_release_38/lib/Support/Unix/Signals.inc
+++ macports_release_38/lib/Support/Unix/Signals.inc
@@ -387,6 +387,15 @@ void llvm::sys::PrintStackTraceOnErrorSignal(bool DisableCrashReporting) {
   AddSignalHandler(PrintStackTraceSignalHandler, nullptr);
 
 #if defined(__APPLE__) && defined(ENABLE_CRASH_OVERRIDES)
+
+/* These aren't defined in the 10.4 SDK, so provide them here */
+#ifndef EXC_MASK_CRASH
+#define EXC_MASK_CRASH 0x400
+#endif
+#ifndef MACH_EXCEPTION_CODES
+#define MACH_EXCEPTION_CODES 0x80000000
+#endif
+
   // Environment variable to disable any kind of crash dialog.
   if (DisableCrashReporting || getenv("LLVM_DISABLE_CRASH_REPORT")) {
     mach_port_t self = mach_task_self();
-- 
2.11.0

